src/scripting/game_lua_kernel.cpp

```diff
@@ -622,12 +625,16 @@ int game_lua_kernel::intf_fire_event(lua_State *L, const bool by_id)
 		}
 	}
 
-	if (!lua_isnoneornil(L, pos)) {
-		data.add_child("first", luaW_checkconfig(L, pos));
+	luaW_toconfig(L, pos, data);
+
+	// Support WML names for some common data
+	if(data.has_child("primary_attack")) {
```

Comment:
This seems to be missing a unit test.

Reply:
Github doesn't send any notification when an emoji is added as a response.

What's the confusion? AFAIK all the existing unit tests were passing before the code for handling the `primary_attack` key was added, and they're still all passing afterwards, which suggests there's no unit test for it.

Reply:
Is there another way to test it? I think it's not much more effort than cut&pasting an existing test.

The code being commented on seems to disagree with the emmylua docs about what `first` and `second` are - here they're attacks, but in the emmylua they're units (or locations of units).

Reply:
Okay, I can understand how that's confusing.

The `first` and `second` in the EmmyLua docs are indeed units. The `first` and `second` here are tags of the `data` argument.

In C++, an event handler is passed three pieces of information – two locations, and a config containing arbitrary additional data. The locations are the units. For attack events, the weapons are passed as `[first]` and `[second]` child tags in the config.

So, what code here does is that if the WML table contains a `[primary_attack]` tag, it converts it to the `[first]` tag that the event handler internally expects. Similarly, `[secondary_attack]` is converted to `[second]`. The choice of tags here is based on the tags used in the `[fire_event]` tag, for consistency; the `[first]` and `[second]` tags can be thought of as an internal detail.

Reply:
Ah, ok.

Would it be sensible to remove any `data["first"]` and `data["second"]` tags added by a confused WML or Lua author?